app/scripts/app-init.js

```diff
@@ -46,8 +50,26 @@ function importAllScripts() {
   );
 }
 
-// Placing script import call here ensures that scripts are inported each time service worker is activated.
-importAllScripts();
+// eslint-disable-next-line
+self.oninstall = () => {
+  importAllScripts();
+};
+
+/*
+ * Message event listener below loads script if they are no longer available.
+ * chrome below needs to be replaced by cross-browser object,
+ * but there is issue in importing webextension-polyfill into service worker.
+ * chrome does seems to work in at-least all chromium based browsers
+ */
+// eslint-disable-next-line
+chrome.runtime.onMessage.addListener((_1, _2, sendResponse) => {
+  // eslint-disable-next-line
+  if (!self.scriptLoaded) {
+    importAllScripts();
+  }
+  // Response below if not required but there is an error if message listener does not send response.
```

Comment:
could you explain this a bit? It's not required but there's a non blocking error logged to the console if we don't send a response?

Reply:
I removed that line plz ignore.